src/Workleap.Extensions.Mongo/Indexing/MongoIndexer.cs

```diff
@@ -12,12 +12,14 @@ internal sealed class MongoIndexer : IMongoIndexer
         ?? throw new InvalidOperationException($"Could not find public instance method {nameof(MongoIndexer)}.{nameof(ProcessAsync)}");
 
     private readonly IMongoClientProvider _mongoClientProvider;
+    private readonly IndexDeleter _indexDeleter;
     private readonly IOptionsMonitor<MongoClientOptions> _optionsMonitor;
     private readonly ILoggerFactory _loggerFactory;
 
-    public MongoIndexer(IMongoClientProvider mongoClientProvider, IOptionsMonitor<MongoClientOptions> optionsMonitor, ILoggerFactory loggerFactory)
+    public MongoIndexer(IMongoClientProvider mongoClientProvider, IndexDeleter indexDeleter,  IOptionsMonitor<MongoClientOptions> optionsMonitor, ILoggerFactory loggerFactory)
     {
         this._mongoClientProvider = mongoClientProvider;
+        this._indexDeleter = indexDeleter;
```

Comment:
Can't it be like IndexCreator and  be static? Or atleast have both as dependencies the same.